src-java/nbworker-topology/nbworker-storm-topology/src/test/java/org/openkilda/wfm/topology/nbworker/services/PathsServiceTest.java

```diff
@@ -206,34 +209,45 @@ public void findNPathsByTransitVlanAndNullMaxLatency()
             throws SwitchNotFoundException, RecoverableException, UnroutableFlowException {
         // as max latency param is null LATENCY starategy will be used instead. It means all 500 paths will be returned
         List<PathsInfoData> paths = pathsService.getPaths(SWITCH_ID_1, SWITCH_ID_2, TRANSIT_VLAN, MAX_LATENCY, null,
-                null);
+                null, MAX_PATH_COUNT);
         assertTransitVlanAndLatencyPaths(paths);
     }
 
     @Test
     public void findNPathsByTransitVlanAndLatency()
             throws SwitchNotFoundException, RecoverableException, UnroutableFlowException {
-        List<PathsInfoData> paths = pathsService.getPaths(SWITCH_ID_1, SWITCH_ID_2, TRANSIT_VLAN, LATENCY, null, null);
+        List<PathsInfoData> paths = pathsService.getPaths(SWITCH_ID_1, SWITCH_ID_2, TRANSIT_VLAN, LATENCY,
+                null, null, MAX_PATH_COUNT);
         assertTransitVlanAndLatencyPaths(paths);
     }
 
     @Test
     public void findNPathsByVxlanAndCost()
             throws SwitchNotFoundException, RecoverableException, UnroutableFlowException {
-        List<PathsInfoData> paths = pathsService.getPaths(SWITCH_ID_1, SWITCH_ID_2, VXLAN, COST, null, null);
+        List<PathsInfoData> paths = pathsService.getPaths(SWITCH_ID_1, SWITCH_ID_2, VXLAN, COST, null,
+                null, MAX_PATH_COUNT);
         assertVxlanAndCostPathes(paths);
     }
 
+    @Test
+    public void findNPathsByMaxPathCount()
+            throws UnroutableFlowException, SwitchNotFoundException, RecoverableException {
+        int maxPathCount = 3;
+        List<PathsInfoData> paths = pathsService.getPaths(SWITCH_ID_1, SWITCH_ID_2, VXLAN, COST, null,
+                null, maxPathCount);
+        assertThat(paths.size(), equalTo(maxPathCount));
+    }
+
```

Comment:
do you need to check out of boundaries value like 0 or -1 here in the next unit test or should I?

Reply:
Ok, I have added the validation and unit test for it